examples/hotrod/cmd/driver.go

```diff
@@ -32,15 +32,16 @@ var driverCmd = &cobra.Command{
 	Short: "Starts Driver service",
 	Long:  `Starts Driver service.`,
 	RunE: func(cmd *cobra.Command, args []string) error {
-		logger := log.NewFactory(logger.With(zap.String("service", "driver")))
+		zapLogger := logger.With(zap.String("service", "driver"))
+		logger := log.NewFactory(zapLogger)
 		server := driver.NewServer(
 			net.JoinHostPort(driverOptions.serverInterface, strconv.Itoa(driverOptions.serverPort)),
 			tracing.Init("driver", metricsFactory.Namespace("driver", nil), logger, jAgentHostPort),
 			metricsFactory,
 			logger,
 			jAgentHostPort,
 		)
-		return server.Run()
+		return logError(zapLogger, server.Run())
```

Comment:
Is it worth logging here and returning the same error? Will this lead to duplicate logs up the call stack? Maybe wrapping within another error makes more sense.

Reply:
it does lead to dup logs, but that's better than not getting logs at all, which would be the case since the `all` command calls all servers except `frontend` via `go ...`, so returned errors go nowhere.

Reply:
Right. I meant we could improve the code to wrap the errors each time, but this is all sample code so I don't really mind.